app/scripts/modules/core/src/function/CreateFunctionButton.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+
+import { Application } from 'core/application';
+import { CloudProviderRegistry, ProviderSelectionService } from 'core/cloudProvider';
+import { IFunction } from 'core/domain';
+import { IFunctionUpsertCommand } from 'core/function';
+import { ModalInjector, ReactInjector } from 'core/reactShims';
+import { IModalComponentProps, Tooltip } from 'core/presentation';
+
+export interface IFunctionModalProps extends IModalComponentProps {
+  className?: string;
+  dialogClassName?: string;
+  app: Application;
+  forPipelineConfig?: boolean;
+  functionDef: IFunction;
+  command?: IFunctionUpsertCommand; // optional, when ejecting from a wizard
+  closeModal?(functionCommand: IFunctionUpsertCommand): void; // provided by ReactModal
+  dismissModal?(rejectReason?: any): void; // provided by ReactModal
+}
+
+export interface ICreateFunctionButtonProps {
+  app: Application;
+}
+
+export class CreateFunctionButton extends React.Component<ICreateFunctionButtonProps> {
+  constructor(props: ICreateFunctionButtonProps) {
+    super(props);
+  }
+
+  private createFunction = (): void => {
+    const { skinSelectionService } = ReactInjector;
+    const { app } = this.props;
+
+    ProviderSelectionService.selectProvider(app, 'function').then(selectedProvider => {
+      skinSelectionService.selectSkin(selectedProvider).then(selectedSkin => {
+        const provider = CloudProviderRegistry.getValue(selectedProvider, 'function', selectedSkin);
+        if (provider.CreateFunctionModal) {
+          provider.CreateFunctionModal.show({
+            app: app,
+            application: app,
+            forPipelineConfig: false,
+            function: null,
+            isNew: true,
+          });
+        } else {
+          // TODO
```

Comment:
Remove comment

Reply:
I moved it to componentWillMount.

Will appear in the next push.